casting/converter.go

```diff
@@ -0,0 +1,106 @@
+package casting
+
+import (
+	"reflect"
+	"strconv"
+	"strings"
+)
+
+func CastArrayOfData(inputTypeMap map[string]string, inputNameArray []string, inputDataArray []interface{}) []interface{} {
+	castedColumns := make([]interface{}, len(inputDataArray))
+
+	for i := range inputDataArray {
+		if column_type, ok := inputTypeMap[inputNameArray[i]]; ok {
+			switch strings.ToLower(column_type) {
+			case "string":
+				castedColumns[i] = castToString(inputDataArray[i])
+			case "integer":
+				castedColumns[i] = castToInt(inputDataArray[i])
+			case "float":
+				castedColumns[i] = castToFloat(inputDataArray[i])
+			case "boolean":
+				castedColumns[i] = castToBool(inputDataArray[i])
+			default:
+				castedColumns[i] = inputDataArray[i]
+			}
+		} else {
+			castedColumns[i] = inputDataArray[i]
+		}
+	}
+
+	return castedColumns
+}
+
+func castToString(inputVar interface{}) string {
```

Comment:
In addition to the integration test that you are adding, to increase the coverage, you can also add some unit test here simply calling castTosting(true) etc.. testing each branch

Reply:
Done